test/users/ferguson/histogram/reductionex_class.bad

```diff
@@ -1,7 +1,2 @@
-reductionex_class.chpl:44: internal error: CAL0124 chpl version 1.18.0 pre-release (02d6b31f68)
-
-Internal errors indicate a bug in the Chapel compiler ("It's us, not you"),
-and we're sorry for the hassle.  We would appreciate your reporting this bug -- 
-please see https://chapel-lang.org/bugs.html for instructions.  In the meantime,
-the filename + line number above may be useful in working around the issue.
-
+reductionex_class.chpl:26: error: Illegal use of identifier 'NBUCKETS' from enclosing type
```

Comment:
I do hope that this general pattern (nested type referring to outer type param/type) is allowed in the future. Do we have an issue / story about it? Also I have some concern that fields with the same name as an outer class field will change in behavior once this is implemented, so some of the above patterns would conceivably change. But hopefully they can continue to work.